packages/editor/src/lib/editor/Editor.ts

```diff
@@ -5224,13 +5226,24 @@ export class Editor extends EventEmitter<TLEventMap> {
 				}
 			}
 
-			if (this.isShapeOfType(shape, 'frame')) {
+			if (this.isShapeOfType<TLFrameShape>(shape, 'frame')) {
 				// On the rare case that we've hit a frame (not its label), test again hitInside to be forced true;
 				// this prevents clicks from passing through the body of a frame to shapes behind it.
 
 				// If the hit is within the frame's outer margin, then select the frame
-				const distance = geometry.distanceToPoint(pointInShapeSpace, hitInside)
-				if (Math.abs(distance) <= margin) {
+				const distance = geometry.distanceToPoint(pointInShapeSpace, hitFrameInside)
+				if (
+					hitFrameInside
+						? // On hitInside, the distance will be negative for hits inside
+							// If the distance is positive, check against the outer margin
+							(distance > 0 && distance <= outerMargin) ||
+							// If the distance is negative, check against the inner margin
+							(distance <= 0 && distance > -innerMargin)
+						: // If hitInside is false, then sadly _we do not know_ whether the
+							// point is inside or outside of the shape, so we check against
+							// the max of the two margins
+							distance <= outerMargin
```

Comment:
This comment doesn't match the logic here (no checking against the max of the two margins)

personally i would still pull this out to an `isDistanceWithinMargins(distance, innerMargin, outerMargin, hitInside)` helper given it's complicated enough and i think intended to be the same here and below

Reply:
fixed comments (and a code issue). I'm not going to create a helper fn since I'm not sure it's exactly the same logic as elsewhere.